zerver/views/realm.py

```diff
@@ -16,7 +16,39 @@
 from zerver.lib.validator import check_string, check_dict, check_bool, check_int
 from zerver.lib.streams import access_stream_by_id
 from zerver.models import Realm, UserProfile
+from zerver.lib.name_restrictions import is_reserved_subdomain, is_disposable_domain
+from zerver.models import get_realm
+
+def check_realm(request, realm_str):
+    error_strings = {
+        'too short': _("Organization name needs to have length 3 or greater."),
+        'extremal dash': _("Organization name cannot start or end with a '-'."),
+        'bad character': _("Organization name can only have lowercase letters, numbers, and '-'s."),
+        'unavailable': _("Organization name unavailable. Please choose a different one."),
+        'invalid': _("Please enter a valid email address."),
+        'disposable': _("Please use your real email address.")
+    }
 
+    if "@" not in realm_str:
+        return JsonResponse({"message": error_strings['invalid']})
+    realm_str = realm_str.split('@')[1]
+    if is_disposable_domain(realm_str):
+        return JsonResponse({"message": error_strings['disposable']})
+    # subdomain
+    if "." not in realm_str:
+        return JsonResponse({"message": error_strings['invalid']})
+    realm_str = realm_str.split('.')[0]
+    # if not realm_str:
+    #     return JsonResponse({"message": error_strings['invalid']})
+    if len(realm_str) < 3:
+        return JsonResponse({"message": error_strings['too short']})
+    if realm_str[0] == '-' or realm_str[-1] == '-':
+        return JsonResponse({"message": error_strings['extremal dash']})
+    if not re.match('^[a-z0-9-]*$', realm_str):
+        return JsonResponse({"message": error_strings['bad character']})
+    if is_reserved_subdomain(realm_str) or get_realm(realm_str) is not None:
+        return JsonResponse({"message": error_strings['unavailable']})
+    return JsonResponse({"message": "OK"})
```

Comment:
This seems to be a duplicate of the logic in `zerver/forms.py`.  I just extracted it as a function in
7445493fb35d95945865c115f30e30aed928675c

You should be able to do e.g.

try:
    check_subdomain_available(subdomain)
except ValidationError:
    return json_error(...)